pkg/distributor/ha_tracker.go

```diff
@@ -113,16 +120,24 @@ func (cfg *HATrackerConfig) RegisterFlags(f *flag.FlagSet) {
 func newClusterTracker(cfg HATrackerConfig) (*haTracker, error) {
 	codec := codec.Proto{Factory: ProtoReplicaDescFactory}
 
-	if cfg.FailoverTimeout <= cfg.UpdateTimeout {
-		return nil, fmt.Errorf("HA Tracker failover timeout must be greater than update timeout, %d is <= %d", cfg.FailoverTimeout, cfg.UpdateTimeout)
+	var jitter time.Duration
+	if cfg.UpdateTimeoutJitterMax != 0 {
+		jitter = time.Duration(rand.Int63n(int64(2*cfg.UpdateTimeoutJitterMax))) - cfg.UpdateTimeoutJitterMax
 	}
+
+	if cfg.FailoverTimeout <= cfg.UpdateTimeout+cfg.UpdateTimeoutJitterMax+time.Second {
+		return nil, fmt.Errorf("HA Tracker failover timeout must be 1s greater than update timeout, %d is <= %d",
```

Comment:
You can make the message more clear this way:

```
fmt.Errorf("HA Tracker failover timeout (%v) must be at least 1s greater than update timeout (%v)", cfg.FailoverTimeout, cfg.UpdateTimeout+cfg.UpdateTimeoutJitterMax)
```

Moreover, given the error message, the check should be `<` instead of `<=`, so:

```
if cfg.FailoverTimeout < cfg.UpdateTimeout+cfg.UpdateTimeoutJitterMax+time.Second {
```